webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -157,10 +171,41 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
       throw new RuntimeException(msg, se);
     }
 
-    Class<O> jobOutputType = getJobOutputType(quartzJobInfo);
     return new QuartzPollableFutureTask<O>(pollableTask, jobOutputType);
   }
 
+  private void skipPendingPollablesWithMatchingId(
+      Scheduler scheduler, JobKey jobKey, Trigger trigger, PollableTask pollableTask) {
```

Comment:
I was under the impression this code could be removed? why do have to refetch the trigger using the jobkey vs. just using the TriggerKey? 
Can the `scheduler.checkExists(triggerKey)` be replaced with `scheduler.getTrigger(triggerKey)` and just read the pollableTask data from there?

Reply:
I don't think we can do this as it's possible for a job to have multiple triggers of the same key, just that these triggers will be in different states. We're only interested in the blocked triggers, so need to find them all and skip the pollable tasks when the criteria is met